applications/tari_swarm_daemon/src/main.rs

```diff
@@ -0,0 +1,309 @@
+        // mining and registrations, so a given wallet is not guaranteed to have funds. There is no big need to fix
+        // at the moment this as we typically only need one wallet.
+        InstanceConfig::new(InstanceType::MinoTariConsoleWallet)
+            .with_name("Minotari Wallet")
+            .with_num_instances(1),
+        // Let's mine 10 blocks on startup by default.
+        InstanceConfig::new(InstanceType::MinoTariMiner)
+            .with_name("Minotari Miner")
+            .with_arg("max_blocks", "10"),
+        InstanceConfig::new(InstanceType::TariValidatorNode)
+            .with_name("Validator node")
+            .with_num_instances(1),
+        InstanceConfig::new(InstanceType::TariIndexer).with_name("Indexer"),
+        InstanceConfig::new(InstanceType::TariWalletDaemon).with_name("Wallet Daemon"),
+        InstanceConfig::new(InstanceType::TariSignalingServer).with_name("Signaling server"),
+    ];
+
+    let base_dir = cli
+        .common
+        .base_dir
+        .clone()
+        .or_else(|| {
+            cli.get_config_path()
+                .canonicalize()
+                .ok()
+                .and_then(|p| p.parent().map(|p| p.to_path_buf()))
+        })
+        .unwrap_or_else(|| std::env::current_dir().unwrap());
+
+    Ok(Config {
+        network: cli.common.network.unwrap_or(Network::LocalNet),
+        start_port: 12000,
+        base_dir: base_dir
+            .canonicalize()
+            .with_context(|| anyhow!("Base path '{}' does not exist", base_dir.display()))?,
+        webserver: WebserverConfig::default(),
+        processes: ProcessesConfig {
+            force_compile: false,
+            instances,
+            executables,
+        },
+    })
+}
+
+async fn start(cli: &Cli) -> anyhow::Result<()> {
+    let mut config = Config::load_with_cli(cli).await?;
+    if let Commands::Start(ref overrides) = cli.command {
+        overrides.apply(&mut config)?;
+    }
+    let _pid = lockfile::Lockfile::create(config.base_dir.join("tari_swarm.pid"))
+        .context("Failed to acquire lockfile. Is another instance already running?")?;
+
+    create_paths(&config).await?;
+
+    let shutdown = Shutdown::new();
+    let signal = shutdown.to_signal().select(exit_signal()?);
+    let (task_handle, pm_handle) = process_manager::spawn(&config, shutdown.to_signal());
+    let webserver = webserver::spawn(config, shutdown.to_signal(), pm_handle.clone());
+
+    tokio::select! {
+        _ = signal => {
+            log::info!("Terminating all instances...");
+            let num_instances = pm_handle.terminate_all().await?;
+            log::info!("Terminated {num_instances} instances");
+        },
+        result = webserver => {
+            result??;
+            log::info!("Webserver exited");
+        },
+        result = task_handle => {
+            result??;
+            log::info!("Process manager exited");
+        }
+    }
+
```

Comment:
do you need to call `pm_handle.terminate_all()` if the second or third branches happened?

Reply:
Yeah you are right for the web server (though I enabled kill on drop which will hard terminate the processes). If the process manager exits we cannot make that call, so just added it for the webserver.